### PR TITLE
Move DEV only module requires into __DEV__ blocks

### PR DESCRIPTION
**Pull request by trueadm**
This is related to #10176. Dangling `require()` calls were created from the build due to the usage not being in a `__DEV__` block within the source code. This PR aims to move the require calls so they no longer dangle in the production builds.

Also, the need to have single quotes within `modulesToStub` blocks has been removed – it was error prone.

### CODE REVIEW

**Comment by gaearon** [comment]
Do you mind rebasing this?

**Comment by trueadm** [comment]
@gaearon Done just now, please give it another check when you have time :) thanks!

File: src/isomorphic/children/traverseAllChildren.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ var REACT_ELEMENT_TYPE =
 
 if (__DEV__) {
   var warning = require('fbjs/lib/warning');
-  var {getCurrentStackAddendum} = require('ReactComponentTreeHook');
+  var {getStackAddendum} = require('ReactDebugCurrentFrame');
```
Comment (gaearon) [inline src/isomorphic/children/traverseAllChildren.js]: This looks like code change?

Comment (gaearon) [inline src/isomorphic/children/traverseAllChildren.js]: Oops nevermind.

**Comment by gaearon** [comment]
Let's get this in. It's not a long term solution but at least it's consistent with what we already do.
Longer term we'll want https://github.com/facebook/react/issues/10176#issuecomment-315398546. We can look at it with ES modules transition.

**Comment by trueadm** [comment]
Cool :) Thanks

**Comment by abigsmall** [comment]
@trueadm I noticed that in some files there remain invocations of `warning` that aren’t explicitly wrapped in `if (__DEV__)` blocks, such as for [`getTouchIdentifier` in `ResponderTouchHistoryStore.js`](https://github.com/facebook/react/blob/357925a84e34d17b96568e67d30baeb2a399da11/src/renderers/shared/shared/event/eventPlugins/ResponderTouchHistoryStore.js#L107): is that something to be concerned about?

Sorry if commenting on already merged PRs is a not recommended, figured I’d try this rather than create a new GitHub issue

**Pull request merged**